### PR TITLE
Added feature to cache user's progress province level at local storage 

### PR DESCRIPTION
**Pull request by hateryx**
Hi,

It's a pleasure to learn about this open source project. Please allow me to contribute a minor feature which aims to increase UX.

Opportunity for Improvement: User's progress on province levels should be saved for his/her next session. 

Proposed Solution: Save the province level at the user's browser local storage via handleLevelClick. In the next user's session, the stored province levels, if any, are retrieved from the local storage and then set as initial state. If no stored province levels, the levelArr is still used.

I hope you consider my PR. Thanks!

### CODE REVIEW

**Comment by warengonzaga** [comment]
hey @hateryx thank you for the PR, basically make the progress persistent? cc: @denzdelvillar @jofftiquez

**Comment by jofftiquez** [comment]
> hey @hateryx thank you for the PR, basically make the progress persistent? cc: @denzdelvillar @jofftiquez 

Sounds good, altho I don't see special use-case. But it wouldn't hurt to have it. Will review within the day. Thanks for the PR

File: src/pages/Map.js [hunk 1]
```diff
@@ -1,28 +1,34 @@
-import React, { useMemo, useState, useEffect, useCallback, useContext } from 'react';
-import { useSearchParams, useNavigate } from 'react-router-dom';
+import React, {
+  useMemo,
+  useState,
+  useEffect,
+  useCallback,
+  useContext,
+} from "react";
```
Comment (jofftiquez) [inline src/pages/Map.js:7]: Please don't replace all `'` with `"`

**Review by jofftiquez — CHANGES_REQUESTED** [review]

File: src/pages/Map.js (hunk 1 above)
Comment (hateryx) [reply src/pages/Map.js]: My bad, I overlooked the changes from the lint. Edited already the file to present the changes that matters. Thanks

File: src/pages/Map.js [hunk 2]
```diff
@@ -37,7 +37,16 @@ const PhilippinesMap = () => {
   useEffect(() => {
     const levelStrFromURL = searchParams.get('levels');
     const levelArr = levelStringToArray(levelStrFromURL);
-    setProvinceLevels(levelArr);
+
+    // Retrieve the provinceLevels array from local storage
+    const storedProvinceLevels = localStorage.getItem("provinceLevels");
```
Comment (jofftiquez) [inline src/pages/Map.js:42]: Please use single quote, thanks.

Comment (jofftiquez) [reply src/pages/Map.js:42]: Apply to all changes

**Review by jofftiquez — CHANGES_REQUESTED** [review]

File: src/pages/Map.js (hunk 2 above)
Comment (hateryx) [reply src/pages/Map.js]: Ok, done already. Thanks!

**Review by jofftiquez — APPROVED** [review]

**Comment by jofftiquez** [comment]
@denzdelvillar kindly review and merge if ok with you thanks

**Review by warengonzaga — APPROVED** [review]
LGTM 🍻

**Comment by jofftiquez** [comment]
@denzdelvillar following up on this one as well. Thanks

**Comment by denzdelvillar** [comment]
code-wise, looks good to me. @warengonzaga any idea why vercel doesn't auto deploy this branch? 🤔

**Comment by warengonzaga** [comment]
@jofftiquez take a look into our branch protection not sure about the setup in there. We are experiencing the same issue here: https://github.com/orgs/community/discussions/26698 cc: @denzdelvillar

**Comment by warengonzaga** [comment]
Fixed @denzdelvillar @jofftiquez

**Pull request merged**